### ISSUE TITLE
Add support for custom configuration

### ISSUE THREAD

**Issue by davidkretch** [paws-r/paws#29]
e.g. we should be able to set endpoints, etc. something like

```r
config(
  Endpoint = "https://example.com"
)
```

**Comment by davidkretch** [comment]
Addressed by #184

**Comment by davidkretch** [comment]
```r
a <- paws::ec2(config = list(region = "us-east-2"))
b <- paws::ec2(config = list(region = "us-east-1"))

# Returns the same list of key pairs.
a$describe_key_pairs()
b$describe_key_pairs()
```

Custom config code is on branch `feature/custom_endpoint`.

**Comment by davidkretch** [comment]
Addressed by #205.
